Tasks/DotNetCoreInstallerV1/versionFetcher.ts

```
@@ -0,0 +1,273 @@
+            }
+            else {
+                let scriptPath = path.join(this.getCurrentDir(), 'externals', 'get-os-distro.sh');
+                this.setFileAttribute(scriptPath, "777");
+
+                scriptRunner = tl.tool(tl.which(scriptPath, true));
+            }
+
+            let result: trm.IExecSyncResult = scriptRunner.execSync();
+
+            if (result.code != 0) {
+                throw tl.loc("getMachinePlatformFailed", result.error ? result.error.message : result.stderr);
+            }
+
+            let output: string = result.stdout;
+
+            let index;
+            if ((index = output.indexOf("Primary:")) >= 0) {
+                let primary = output.substr(index + "Primary:".length).split(os.EOL)[0];
+                osSuffix.push(primary);
+                console.log(tl.loc("PrimaryPlatform", primary));
+            }
+
+            if ((index = output.indexOf("Legacy:")) >= 0) {
+                let legacy = output.substr(index + "Legacy:".length).split(os.EOL)[0];
+                osSuffix.push(legacy);
+                console.log(tl.loc("LegacyPlatform", legacy));
+            }
+
+            if (osSuffix.length == 0) {
+                throw tl.loc("CouldNotDetectPlatform");
+            }
+        }
+        catch (ex) {
+            throw tl.loc("FailedInDetectingMachineArch", ex)
+        }
+
+        return osSuffix;
+    }
+
+    private setFileAttribute(file: string, mode: string): void {
+        fs.chmodSync(file, mode);
+    }
+
+    private getCurrentDir(): string {
+        return __dirname;
+    }
+
+    private releasesIndex: any;
+    private httpCallbackClient: httpClient.HttpClient;
+}
+
+export class VersionInfo {
+    public version: string;
+    public files: VersionFilesData[];
+
+    public static getRuntimeVersion(versionInfo: VersionInfo): string {
+        if (versionInfo["runtime-version"]) {
+            return versionInfo["runtime-version"];
+        }
+
+        return versionInfo.version;
```

why are we returing this? Should we throw exception if the above field is not found

---

This is to handle both cases 
1. `Package Type = sdk` : in this case the object contains `runtime-version` and that will be returned. This is needed to create completeFile for runtime installed along with sdk.
2. `PackageType = runtime`: in this case `version` itself is runtime-version, and thus that should be returned.

I feel that if runtime-version is not found in sdk object, we can skip creating the runtime complete file and give a warning. Changing code to behave the same.
@sachinma  thoughts ?